File: content-solutions/stack/taller-neighbor.java
import java.util.*;

class Solution {
    public int[] solution(int[] heights) {
        int[] answer = new int[heights.length];
        Deque<Integer> stack = new ArrayDeque<>();
        for (int i = 0; i < heights.length; i++) {
            while (!stack.isEmpty() && heights[stack.peek()] <= heights[i]) stack.pop();
            answer[i] = stack.isEmpty() ? 0 : stack.peek() + 1;
            stack.push(i);
        }
        return answer;
    }
}

File: content-solutions/stack/taller-neighbor.py
def solution(heights):
    answer = [0] * len(heights)
    stack = []
    for i, h in enumerate(heights):
        while stack and heights[stack[-1]] <= h:
            stack.pop()
        answer[i] = stack[-1] + 1 if stack else 0
        stack.append(i)
    return answer
